### PR TITLE
Introduce configmap in 9c-main

### PR DESCRIPTION
**Pull request by Atralupus**

### CODE REVIEW

File: 9c-main/remote-headless-1.yaml [hunk 1]
```diff
@@ -75,6 +75,16 @@ spec:
...
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
```
Comment (area363) [inline 9c-main/remote-headless-1.yaml:87]:
```suggestion
```

File: 9c-main/remote-headless-2.yaml [hunk 2]
```diff
@@ -75,6 +75,16 @@ spec:
...
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
```
Comment (area363) [inline 9c-main/remote-headless-2.yaml:87]:
```suggestion
```

File: 9c-main/snapshot-partition-reset.yaml [hunk 3]
```diff
@@ -56,7 +56,7 @@ spec:
             - mountPath: /data
               name: snapshot-volume-reset
           - args:
-            - 100302/6ec8E598962F1f475504F82fD5bF3410eAE58B9B/MEUCIQCiXJnTqI.XvSOo.YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
+            - $(APP_PROTOCOL_VERSION_KEY).YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
```
Comment (area363) [inline 9c-main/snapshot-partition-reset.yaml:59]:
```suggestion
            - $(APP_PROTOCOL_VERSION_KEY)
```

File: 9c-main/snapshot-partition-reset.yaml [hunk 4]
```diff
@@ -78,9 +78,14 @@ spec:
...
                   key: token
+            - name: APP_PROTOCOL_VERSION_KEY
+              valueFrom:
+                configMapKeyRef:
+                  name: version-config
+                  key: APP_PROTOCOL_VERSION
           - args:
             - temp
-            - 100302/6ec8E598962F1f475504F82fD5bF3410eAE58B9B/MEUCIQCiXJnTqI.XvSOo.YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
+            - $(APP_PROTOCOL_VERSION_KEY).YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
```
Comment (area363) [inline 9c-main/snapshot-partition-reset.yaml:88]:
```suggestion
            - $(APP_PROTOCOL_VERSION_KEY)
```

File: 9c-main/snapshot-partition-reset.yaml [hunk 5]
```diff
@@ -103,13 +108,18 @@ spec:
...
+                  name: version-config
+                  key: APP_PROTOCOL_VERSION
             - name: CF_DISTRIBUTION_ID
               valueFrom:
                 secretKeyRef:
                   name: aws-keys
                   key: cf_distribution_id
           - args:
-            - 100302/6ec8E598962F1f475504F82fD5bF3410eAE58B9B/MEUCIQCiXJnTqI.XvSOo.YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
+            - $(APP_PROTOCOL_VERSION_KEY).YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
```
Comment (area363) [inline 9c-main/snapshot-partition-reset.yaml:122]:
```suggestion
            - $(APP_PROTOCOL_VERSION_KEY)
```

File: 9c-main/snapshot-partition-reset.yaml [hunk 6]
```diff
@@ -131,9 +141,14 @@ spec:
...
                   key: token
+            - name: APP_PROTOCOL_VERSION_KEY
+              valueFrom:
+                configMapKeyRef:
+                  name: version-config
+                  key: APP_PROTOCOL_VERSION
           - args:
             - temp
-            - 100302/6ec8E598962F1f475504F82fD5bF3410eAE58B9B/MEUCIQCiXJnTqI.XvSOo.YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
+            - $(APP_PROTOCOL_VERSION_KEY).YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
```
Comment (area363) [inline 9c-main/snapshot-partition-reset.yaml:151]:
```suggestion
            - $(APP_PROTOCOL_VERSION_KEY)
```

File: 9c-main/snapshot-partition.yaml [hunk 7]
```diff
@@ -12,7 +12,7 @@ spec:
         spec:
           containers:
           - args:
-            - 100302/6ec8E598962F1f475504F82fD5bF3410eAE58B9B/MEUCIQCiXJnTqI.XvSOo.YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
+            - $(APP_PROTOCOL_VERSION_KEY).YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
```
Comment (area363) [inline 9c-main/snapshot-partition.yaml:15]:
```suggestion
            - $(APP_PROTOCOL_VERSION_KEY)
```

File: 9c-main/snapshot-partition.yaml [hunk 8]
```diff
@@ -35,14 +35,19 @@ spec:
...
+                  key: APP_PROTOCOL_VERSION
             - name: CF_DISTRIBUTION_ID
               valueFrom:
                 secretKeyRef:
                   name: aws-keys
                   key: cf_distribution_id
           initContainers:
           - args:
-            - 100302/6ec8E598962F1f475504F82fD5bF3410eAE58B9B/MEUCIQCiXJnTqI.XvSOo.YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
+            - $(APP_PROTOCOL_VERSION_KEY).YbjR13jespoTcLnv25Mf6KmoYomlwIgZKFqAChtennyviPM.TLFgsjZCwu9gM.wpN78.nMN7Us=/ZHU4OmxhdW5jaGVydTQyOjEvNTBkYTNhZjNlYmM3OGFlN2ZjMGZiOTFlY2M4MzM5ZmQ0MGQxYTkyZXU2OnBsYXllcnU0MjoxLzc4MWVkMTQ0YWJlYTQ4MmU1ODFkNjM5MWIwNDUyMzI5MmU3MzI2NjF1OTp0aW1lc3RhbXB1MTA6MjAyMi0xMC0wNGU=
```
Comment (area363) [inline 9c-main/snapshot-partition.yaml:50]:
```suggestion
            - $(APP_PROTOCOL_VERSION_KEY)
```

File: 9c-main/full-state.yaml [hunk 9]
```diff
@@ -62,6 +62,11 @@ spec:
         - --chain-tip-stale-behavior-type=reboot
         command:
         - dotnet
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
```
Comment (area363) [inline 9c-main/full-state.yaml:69]:
```suggestion
        env:
        - name: APP_PROTOCOL_VERSION_KEY
          valueFrom:
            configMapKeyRef:
              name: version-config
              key: APP_PROTOCOL_VERSION
```

File: 9c-main/remote-headless-31.yaml [hunk 10]
```diff
@@ -64,6 +64,11 @@ spec:
         - --chain-tip-stale-behavior-type=reboot
         command:
         - dotnet
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
```
Comment (area363) [inline 9c-main/remote-headless-31.yaml:71]:
```suggestion
        env:
        - name: APP_PROTOCOL_VERSION_KEY
          valueFrom:
            configMapKeyRef:
              name: version-config
              key: APP_PROTOCOL_VERSION
```

File: 9c-main/remote-headless-99.yaml [hunk 11]
```diff
@@ -64,6 +64,11 @@ spec:
         - --chain-tip-stale-behavior-type=reboot
         command:
         - dotnet
+        - name: APP_PROTOCOL_VERSION_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: version-config
+              key: APP_PROTOCOL_VERSION
```
Comment (area363) [inline 9c-main/remote-headless-99.yaml:71]:
```suggestion
        env:
        - name: APP_PROTOCOL_VERSION_KEY
          valueFrom:
            configMapKeyRef:
              name: version-config
              key: APP_PROTOCOL_VERSION
```

**Comment by area363** [comment]
I think we should also add a `configmap-versions.yaml` file like https://github.com/planetarium/9c-k8s-config/blob/main/9c-internal/configmap-versions.yaml and apply it in `deploy-main.sh` perhaps before this line(https://github.com/planetarium/9c-k8s-config/blob/main/9c-main/deploy-main.sh#L52).

File: 9c-main/remote-headless-99.yaml (hunk 11 above)
Comment (Atralupus) [reply 9c-main/remote-headless-99.yaml]: Done! thanks

File: 9c-main/remote-headless-31.yaml (hunk 10 above)
Comment (Atralupus) [reply 9c-main/remote-headless-31.yaml]: I moved `APP_PROTOCOL_VERSION_KEY` under the `env` part, not created

File: 9c-main/full-state.yaml (hunk 9 above)
Comment (Atralupus) [reply 9c-main/full-state.yaml:70]: Done!

File: 9c-main/snapshot-partition.yaml (hunk 8 above)
Comment (Atralupus) [reply 9c-main/snapshot-partition.yaml]: Done!

File: 9c-main/snapshot-partition.yaml (hunk 7 above)
Comment (Atralupus) [reply 9c-main/snapshot-partition.yaml]: Done!

File: 9c-main/snapshot-partition-reset.yaml (hunk 6 above)
Comment (Atralupus) [reply 9c-main/snapshot-partition-reset.yaml]: Done!

File: 9c-main/snapshot-partition-reset.yaml (hunk 5 above)
Comment (Atralupus) [reply 9c-main/snapshot-partition-reset.yaml]: Done!

File: 9c-main/snapshot-partition-reset.yaml (hunk 4 above)
Comment (Atralupus) [reply 9c-main/snapshot-partition-reset.yaml]: Done!

File: 9c-main/snapshot-partition-reset.yaml (hunk 3 above)
Comment (Atralupus) [reply 9c-main/snapshot-partition-reset.yaml]: uh... I missed replacing it thanks

File: 9c-main/remote-headless-2.yaml (hunk 2 above)
Comment (Atralupus) [reply 9c-main/remote-headless-2.yaml]: Done!

File: 9c-main/remote-headless-1.yaml (hunk 1 above)
Comment (Atralupus) [reply 9c-main/remote-headless-1.yaml]: @.@ duplicated sorry

**Comment by Atralupus** [comment]
> I think we should also add a `configmap-versions.yaml` file like https://github.com/planetarium/9c-k8s-config/blob/main/9c-internal/configmap-versions.yaml and apply it in `deploy-main.sh` perhaps before this line(https://github.com/planetarium/9c-k8s-config/blob/main/9c-main/deploy-main.sh#L52).

Finish and thanks for your detail

**Comment by area363** [comment]
Thanks for the update! By the way, we still have to add a `configmap-versions.yaml` file under the `9c-main` directory. 🙏 

Also, for this failing test(https://github.com/planetarium/9c-k8s-config/actions/runs/3231521671/jobs/5291138735#step:5:13), should we change this code: https://github.com/planetarium/9c-k8s-config/blob/6e978d85f06bc335f51eb19d05ad141c262350d1/.github/bin/check_apv_consistency.py#L70 to `return env["valueFrom"]["configMapKeyRef"]["key"]` to match the PR(https://github.com/planetarium/9c-k8s-config/pull/21/files#diff-65717de434291c9d00622c6ff5602894b5232ea0c35a6647f5b2f8842d9700daR31-R34)?

Or should we just modify `check_apv_consistency.py` to take take the value from `configmap-versions.yaml`? Let me know your thoughts!

**Comment by Atralupus** [comment]
> Thanks for the update! By the way, we still have to add a `configmap-versions.yaml` file under the `9c-main` directory. 🙏
> 
> Also, for this failing test(https://github.com/planetarium/9c-k8s-config/actions/runs/3231521671/jobs/5291138735#step:5:13), should we change this code:
> 
> https://github.com/planetarium/9c-k8s-config/blob/6e978d85f06bc335f51eb19d05ad141c262350d1/.github/bin/check_apv_consistency.py#L70
> 
> to `return env["valueFrom"]["configMapKeyRef"]["key"]` to match the PR(https://github.com/planetarium/9c-k8s-config/pull/21/files#diff-65717de434291c9d00622c6ff5602894b5232ea0c35a6647f5b2f8842d9700daR31-R34)?
> Or should we just modify `check_apv_consistency.py` to take take the value from `configmap-versions.yaml`? Let me know your thoughts!

We already have [configmap-versions](https://github.com/planetarium/9c-k8s-config/blob/main/9c-main/configmap-versions.yaml)
Now, I think we don't have a reason to check apv, So I delete the check apv script

**Comment by area363** [comment]
> > Thanks for the update! By the way, we still have to add a `configmap-versions.yaml` file under the `9c-main` directory. 🙏
> > Also, for this failing test(https://github.com/planetarium/9c-k8s-config/actions/runs/3231521671/jobs/5291138735#step:5:13), should we change this code:
> > https://github.com/planetarium/9c-k8s-config/blob/6e978d85f06bc335f51eb19d05ad141c262350d1/.github/bin/check_apv_consistency.py#L70
> > 
> > to `return env["valueFrom"]["configMapKeyRef"]["key"]` to match the PR(https://github.com/planetarium/9c-k8s-config/pull/21/files#diff-65717de434291c9d00622c6ff5602894b5232ea0c35a6647f5b2f8842d9700daR31-R34)?
> > Or should we just modify `check_apv_consistency.py` to take take the value from `configmap-versions.yaml`? Let me know your thoughts!
> 
> We already have [configmap-versions](https://github.com/planetarium/9c-k8s-config/blob/main/9c-main/configmap-versions.yaml) Now, I think we don't have a reason to check apv, So I delete the check apv script

Oops, I didn't realize `configmap-versions.yaml` was already in. My bad 😅 
Ok, deleting the check script makes sense. LGTM!

**Review by area363 — APPROVED** [review]

**Review by sky1045 — APPROVED** [review]

**Pull request merged**